src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IfError.cs

```diff
@@ -96,77 +96,80 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             var count = args.Length;
             nodeToCoercedTypeMap = null;
 
-            // Check the predicates.
             var fArgsValid = true;
-            var type = ReturnType;
 
-            var isBehavior = context.AllowsSideEffects;
+            /*
+            If we were to write IfError(v1, f1, v2, f2, ..., vn, fn) in a try...catch way, it would look something like:
+                
+            var result
+            try {
+                result = v1();
+            } catch {
+                return f1();
+            }
+            try {
+                result = v2();
+            } catch {
+                return f2();
+            }
+            ...
+            try {
+                result = vn();
+            } catch {
+                return fn();  // omit this last catch if there is an odd number of arguments
+            }
+            return result;
 
-            Contracts.Assert(type == DType.Unknown);
-            for (var i = 0; i < count;)
+            In this case, we prefer the type of the last v_i as the return type.
+            */
+
+            var preferredTypeIndex = (count % 2) == 0 ? count - 2 : count - 1;
+            var type = argTypes[preferredTypeIndex];
+            if (type.IsError)
+            {
+                errors.EnsureError(args[preferredTypeIndex], TexlStrings.ErrTypeError);
+                fArgsValid = false;
+            }
+
+            for (var i = count - 1; i >= 0; i -= 2)
```

Comment:
>for (var i = count - 1; i >= 0; i -= 2) [](http://example.com/codeflow?start=12&length=39)

Why going backwards, when most functions go forwards? Not a problem with this PR, just curiosity :-)

Reply:
It is so,
In Case of last arg is Blank() or error, I can have a return type from last possible arg that can be returned.
e.g. 
even arg `If(v1, f1, Blank(), f2)` return type would be same as f2. and if f2 is blank then of f1 and so on...
odd arg `If(v1, f1, Blank())` return type would be same as f1.